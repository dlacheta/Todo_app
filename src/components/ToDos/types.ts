export type FormDataType = {
  todo: string;
};
