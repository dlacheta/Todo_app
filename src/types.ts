export type TodosType = {
  id: string;
  todo: string;
  completed: boolean;
};

export enum FilterEnum {
  all,
  active,
  completed,
}
