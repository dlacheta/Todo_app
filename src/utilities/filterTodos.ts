import { TodosType, FilterEnum } from '../types';

const filterTodos = (todos: TodosType[], filter: FilterEnum): TodosType[] => {
  if (filter === FilterEnum.active)
    return todos.filter((todo) => !todo.completed);
  if (filter === FilterEnum.completed)
    return todos.filter((todo) => todo.completed);
  return todos;
};

export default filterTodos;
