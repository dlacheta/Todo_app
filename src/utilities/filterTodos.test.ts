import filterTodos from './filterTodos';
import { FilterEnum } from '../types';

const todos = [
  {
    todo: 'laundry',
    completed: false,
  },
  {
    todo: 'buy milk',
    completed: true,
  },
  {
    todo: 'buy eggs',
    completed: false,
  },
];

test('should return all todos', () => {
  const result = filterTodos(todos, FilterEnum.all);

  expect(result).toMatchSnapshot();
});

test('should return all active todos', () => {
  const result = filterTodos(todos, FilterEnum.active);

  expect(result).toMatchSnapshot();
});

test('should return all completed todos', () => {
  const result = filterTodos(todos, FilterEnum.completed);

  expect(result).toMatchSnapshot();
});
