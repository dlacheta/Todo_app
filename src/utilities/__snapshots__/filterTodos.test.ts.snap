// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return all active todos 1`] = `
Array [
  Object {
    "completed": false,
    "todo": "laundry",
  },
  Object {
    "completed": false,
    "todo": "buy eggs",
  },
]
`;

exports[`should return all completed todos 1`] = `
Array [
  Object {
    "completed": true,
    "todo": "buy milk",
  },
]
`;

exports[`should return all todos 1`] = `
Array [
  Object {
    "completed": false,
    "todo": "laundry",
  },
  Object {
    "completed": true,
    "todo": "buy milk",
  },
  Object {
    "completed": false,
    "todo": "buy eggs",
  },
]
`;
